internal/services/resource/resource_provider_registration_resource_test.go

```diff
@@ -123,6 +132,65 @@ resource "azurerm_resource_provider_registration" "import" {
 `, template)
 }
 
+func unRegisterProvider() error {
+	client, err := testclient.Build()
+	if err != nil {
+		return fmt.Errorf("building client: %+v", err)
+	}
+
+	ctx, cancel := context.WithDeadline(client.StopContext, time.Now().Add(30*time.Minute))
+	defer cancel()
+
+	const resourceProvider = "Microsoft.ApiSecurity"
```

Comment:
Can we make this a function argument (and handle multiple RP names) so that it's less opaque what to do if we later update the tests?

e.g.

```golang
func (ResourceProviderRegistrationResource) unRegisterProvider(resourceProviders ...string) error {
```